### ISSUE TITLE
Violation of Reactive Streams rule 3.16

### ISSUE THREAD

**Issue by rossabaker** [http4s/http4s#719]
```
ERROR] 2016-09-20 09:07:09,022  org.http4s.client.asynchttpclient.UnicastSubscriber request - null violated the Reactive Streams rule 3.16 by throwing an exception from request.
java.lang.NullPointerException
    at org.http4s.client.asynchttpclient.UnicastSubscriber.request(UnicastSubscriber.scala:89)
    at org.http4s.client.asynchttpclient.AsyncHttpClient$$anon$2$$anon$1.request(AsyncHttpClient.scala:91)
    at org.http4s.client.asynchttpclient.QueueSubscriber.org$http4s$client$asynchttpclient$QueueSubscriber$$$anonfun$1(QueueSubscriber.scala:19)
    at org.http4s.client.asynchttpclient.QueueSubscriber$$Lambda$120/1853720677.apply$mcV$sp(Unknown Source)
```

I've heard a couple reports of this, but never successfully reproduced it.  Maybe [https://github.com/http4s/http4s/issues/710#issuecomment-248155074] will help.

**Comment by ericmann** [comment]
I've also seen this exact error with some requests I've made using the AsyncHttpClient (specifically trying to code a "send and forget" request for a pubsub system).

**Comment by rossabaker** [comment]
@ericmann On which version?  I made some changes that I thought might have fixed this.

**Comment by rossabaker** [comment]
I think this is fixed by #1022, and should not be seen starting with 0.15.7.

If anyone sees this on a later version, please reopen.
